### PR TITLE
Update one-page.html to use spectacle 5

### PR DESCRIPTION
**Pull request by VirtualDOMinic**
### Description

- Fixes CSSStyleDeclaration index property setter error.
- Runs prettier on `one-page.html`, too.

Loosely relates to issue #55 - primarily the last comment on there (from @ebello on 17 Apr)

I feel the original bug in that issue has likely been resolved by the latest version (not directly confirmed), but the issue raised in the comment is fixed by this.

#### Type of Change

Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

Tried `one-page.html` before and after: 

- Before, the error occurs on `one-page.html#/12` on Chrome 74\* & Firefox 67\*\*
  - Error doesn't occur on Safari
- After, the error doesn't occur on any slide on Chrome, Firefox, Safari.
  - Works fine with both the `one-page.js` from versions `^4` and `^5` as they are identical, but have updated this to version 5 for consistency

\* Error: `TypeError: Failed to set an indexed property on 'CSSStyleDeclaration': Index property setter is not supported.`
\*\* Error: `TypeError: "CSS2Properties doesn't have an indexed property setter for '0'"`

Checklist: complete

### CODE REVIEW

File: one-page.html [hunk 1]
```diff
@@ -1,292 +1,304 @@
...
+                      <Cite>Ken Wheeler</Cite>
+                    </BlockQuote>
+                  </Slide>
+                  <Slide transition={["spin", "zoom"]} bgColor="tertiary">
+                    <Heading caps fit size={1} textColor="primary">
+                      Inline Markdown
+                    </Heading>
+                    <Markdown>
+                      {`
+      ![Markdown Logo](https://cdn.rawgit.com/FormidableLabs/spectacle/master/example/assets/markdown.png)
```
Comment (ryan-roemer) [inline one-page.html]: GLOBAL: The markdown stuff cannot be indented or it won't render. It either has to be deindented like before or use a dedent function something like the following that preserves relative indentation: 

```js
        // Indent to smallest non-empty whitespace level and trim start / end of string.
        const indent = val => {
          let prefix = null;
          return (val || "").split("\n").filter(line => {
            const [cur, remainder] = (line.match(/^([ ]*)([^ ]+)/) || []).slice(1);
            return remainder ? (prefix = null === prefix || cur.length < prefix.length ? cur : prefix,
            !0) : null !== prefix;
          }).map(line => prefix ? line.replace(prefix, "") : line).join("\n").trimRight();
        };
```

Comment (VirtualDOMinic) [reply one-page.html]: Thanks for spotting this! Weirdly whitespace-only diffs didn't show up in my vs code version control tab... and then I glossed over the different look of those markdown slides 🤦‍♂ .

Sorry for a potential over-simplification of the ustility function, but would "sanitising" `Markdown` and `MarkdownSlides` with `.replace(/^[ \t]+/gm, '')` not be sufficient? It seems to work for both of these components in this use case.

If so, would you prefer I add this sanitisation to the components themselves, or try to implement a prettier ignore for these sections instead? IMO the former would be more robust

Comment (VirtualDOMinic) [reply one-page.html]: For clarity, my usage of that `replace()` is in the two spots contained below:

```js 
<Markdown>
                      {`
      ![Markdown Logo](https://cdn.rawgit.com/FormidableLabs/spectacle/master/example/assets/markdown.png)

      You can write inline images, [Markdown Links](http://commonmark.org), paragraph text and most other markdown syntax
      * Lists too!
      * With ~~strikethrough~~ and _italic_
      * And let's not forget **bold**
                      `.replace(/^[ \t]+/gm, '')}
                    </Markdown>
                  </Slide>
                  {
                    MarkdownSlides(`
      #### Create Multiple Slides in Markdown
      All the same tags and elements supported in <Markdown /> are supported in MarkdownSlides.
      ---
      Slides are separated with **three dashes** and can be used _anywhere_ in the deck. The markdown can either be:
      * A Tagged Template Literal
      * Imported Markdown from another file
                    `.replace(/^[ \t]+/gm, ''))
                  }
```

Comment (ryan-roemer) [reply one-page.html]: Yep. That's why code snippet has to do the `prefix` inference and uses the heuristic of "first non-whitespace level of indentation is the indentation to strip of the rest".

Put more concisely, we need to transform:

```md
    * one
    * two
        * a
        * b
```

to this:

```md
* one
* two
    * a
    * b
```

not this:

```md
* one
* two
* a
* b
```

I'm not sure if it belongs in spectacle core or just as a snippet in `one-page.html` itself, but I think it's either that or _don't_ autoformat whitespace which in our case creates a correctness problem since the spacing is meaningful.

That snippet I pasted is pretty battle-honed as I use it in nearly all my one page presentations...

Comment (ryan-roemer) [reply one-page.html]: That said, if we're doing it officially in spectacle (probably don't need for one-page.html because that's up to the user, we should add specific unit tests to cover corner cases like:

- Indentation token to strip occurs again within text and needs to be preserved
- Windows line breaks
- Tabs instead of spaces
- ...etc.

I'm cool with whatever we decide...

Comment (VirtualDOMinic) [reply one-page.html]: Saving this discussion in my bookmarks to take into account during re-write! 

For now, have just undone all the prettier indentation changes inside the `<script>`

**Comment by VirtualDOMinic** [comment]
@ryan-roemer Travis now passing. Let me know if you're happy for this to merge!
